### PR TITLE
Layer 2: Wallet Connect connection with network and account address

### PR DESCRIPTION
**Pull request by estebanmino**
This PR handles Wallet Connect v1 session connections giving users the option to select a different network (including L2) and wallet accounts they want to connect with. 
https://www.figma.com/file/iqIm5yahlws5gClBYX6cZB/connect?node-id=662%3A5

Two things left:

- [x] Display different account addresses under different wallets
- [x] Something is no working with the network logos when changing network on `ContextMenuButton`, that's making the logos not render correctly

Working iOS and Android https://recordit.co/NMmXfG5zCU

### CODE REVIEW

File: src/screens/ChangeWalletSheet.js [hunk 1]
```diff
@@ -50,8 +51,8 @@ import {
 import logger from 'logger';
 
 const deviceHeight = deviceUtils.dimensions.height;
-const footerHeight = 111;
-const listPaddingBottom = 6;
+export const footerHeight = 111;
+export const listPaddingBottom = 6;
```
Comment (estebanmino) [inline src/screens/ChangeWalletSheet.js:55]: remember to remove the exports here

File: src/navigation/Routes.ios.js [hunk 2]
```diff
@@ -411,6 +399,16 @@ function NativeStackNavigator() {
...
+      <NativeStack.Screen
+        component={WalletConnectApprovalSheet}
+        name={Routes.WALLET_CONNECT_APPROVAL_SHEET}
+        {...basicSheetConfig}
+      />
+      <NativeStack.Screen
+        component={WalletConnectRedirectSheet}
+        name={Routes.WALLET_CONNECT_REDIRECT_SHEET}
+        {...basicSheetConfig}
+      />
```
Comment (estebanmino) [inline src/navigation/Routes.ios.js:411]: I had to do this in order to render a `ContextMenuButton` on top of the Sheet. Under `Stack` it wasn't working.

File: src/screens/ChangeWalletSheet.js [hunk 3]
```diff
@@ -132,11 +134,16 @@ export default function ChangeWalletSheet() {
     selectedWallet
   );
 
+  // FIX THIS
+  const watchOnly = false;
```
Comment (estebanmino) [inline src/screens/ChangeWalletSheet.js:138]: remember to send this from the navigation side

File: src/screens/WalletConnectApprovalSheet.js [hunk 4]
```diff
@@ -1,30 +1,41 @@
...
+import { ContactAvatar } from '../components/contacts';
+import ImageAvatar from '../components/contacts/ImageAvatar';
+import { Centered, Column, Row } from '../components/layout';
 import {
   Sheet,
   SheetActionButton,
   SheetActionButtonRow,
 } from '../components/sheet';
 import { Text } from '../components/text';
+import { useTheme } from '@rainbow-me/context';
```
Comment (brunobar79) [inline src/screens/WalletConnectApprovalSheet.js:28]: `useTheme`  is already in the global scope so we don't need to import it on every file 😄

File: src/screens/WalletConnectApprovalSheet.js [hunk 5]
```diff
@@ -37,23 +48,70 @@ const DappLogo = styled(RequestVendorLogoIcon).attrs(
...
+  margin-bottom: 4;
+`;
+
+const LabelText = styled(Text).attrs(() => ({
+  lineHeight: 22,
+  size: 'large',
+  weight: 'heavy',
+}))``;
+
+const AvatarWrapper = styled(View).attrs(() => ({}))`
```
Comment (brunobar79) [inline src/screens/WalletConnectApprovalSheet.js:65]: If you're not using `attrs` there's no need to declare it. You can do this:

```suggestion
const AvatarWrapper = styled(View)`
```

File: src/screens/WalletConnectApprovalSheet.js [hunk 6]
```diff
@@ -37,23 +48,70 @@ const DappLogo = styled(RequestVendorLogoIcon).attrs(
...
+const SwitchText = styled(Text).attrs(() => ({
+  fontSize: 18,
+  lineHeight: 22,
+  weight: 'heavy',
+}))`
+  margin-left: 5;
+`;
+
+export const SavingsSheetEmptyHeight = 313;
+export const SavingsSheetHeight = android ? 424 : 352;
```
Comment (brunobar79) [inline src/screens/WalletConnectApprovalSheet.js:78]: Should we rename these two variables?

File: src/screens/WalletConnectApprovalSheet.js [hunk 7]
```diff
@@ -86,13 +144,69 @@ export default function WalletConnectApprovalSheet() {
...
+      ...approvalAccountInfo,
+      accountLabel:
+        approvalAccountInfo.accountENS ||
+        approvalAccountInfo.accountName ||
+        approvalAccount.address,
+    };
+  }, [walletNames, approvalNetwork, approvalAccount]);
+
+  const approvalNetworkInfo = useMemo(() => {
+    const value = get(networkInfo[approvalNetwork], 'value');
```
Comment (brunobar79) [inline src/screens/WalletConnectApprovalSheet.js:164]: we're trying to get rid of `get` from lodash and use optional chaining instead:
```suggestion
    const value = networkInfo[approvalNetwork]?.value
```

File: src/screens/WalletConnectApprovalSheet.js [hunk 8]
```diff
@@ -86,13 +144,69 @@ export default function WalletConnectApprovalSheet() {
...
+    return {
+      chainId: ethereumUtils.getChainIdFromNetwork(approvalNetwork),
+      color: get(networkInfo[approvalNetwork], 'color'),
```
Comment (brunobar79) [inline src/screens/WalletConnectApprovalSheet.js:167]: same

File: src/screens/WalletConnectApprovalSheet.js [hunk 9]
```diff
@@ -86,13 +144,69 @@ export default function WalletConnectApprovalSheet() {
...
+      name: value.charAt(0).toUpperCase() + value.slice(1),
```
Comment (brunobar79) [inline src/screens/WalletConnectApprovalSheet.js:168]: You can use `capitalize` from `lodash`

**Review by brunobar79 — APPROVED** [review]
🙌 🙌 🙌 

All the comments I've added are very minor, and also there's a linter error that needs to be solved before merging.

File: src/screens/WalletConnectApprovalSheet.js (hunk 6 above)
Comment (estebanmino) [reply src/screens/WalletConnectApprovalSheet.js:78]: oh I just removed these, i forgot to

File: src/screens/WalletConnectApprovalSheet.js (hunk 7 above)
Comment (estebanmino) [reply src/screens/WalletConnectApprovalSheet.js]: cool, I saw it on other places so I guessed that was the way you decided to do it 👌

**Pull request merged**